### ISSUE TITLE
KeyNotFoundException whenever opening magical storage crafting interface

### ISSUE THREAD

**Issue by internetyh** [Ishigh1/RecursiveCraft#9]
Whenever I open the crafting interface from magical storage the following error appears in the chat but doesn't crash the game 

![error](https://user-images.githubusercontent.com/59629341/130413347-3c5b20a5-547b-4eb5-8a2c-170b721e9cae.png)

This error spams the chat until magical storage loads the crafting recipe I searched

Mod List 

1. Magic Storage - Extra
2. Utility Bombs Reborn
3. Zenith Gun
4. Cheat Sheet
5. Recursive Craft
6. No Breaking Doors
7. Recipe Browser
8. Fargo's Crafting Tree Generator
9. Extended Crafting Recipes
10. Smart Doors
11. Spirit Mod
12. Fargo's Souls Mod
13. Luiafk
14. Shadows of Abaddon
15. Thorium Mod
16. VeinMiner
17. Fargo's Mutant Mod
18. Dragon Ball Terraria
19. Calamity Mod
20. Boss Checklist
21. AlchemistNPC Lite

**Comment by Ishigh1** [comment]
Can you send the log file please ?

**Comment by internetyh** [comment]
Where would I find the log file, because this bug seems to happen seemingly randomly (for hours at a time it occurs, and then it stops for hours and then occurs again)

**Comment by Ishigh1** [comment]
After it happened, go to
On Windows: %userprofile%/Documents/My games/Terraria/ModLoader/Logs
On Linux: ~/.local/share/Terraria/ModLoader/Logs
On Mac: ~/Library/Application support/Terraria/ModLoader/Logs
It's client.log

**Comment by internetyh** [comment]
[client.log](https://github.com/Ishigh1/RecursiveCraft/files/7050303/client.log)

It happened as soon as I tried to craft bottled water with no bottles, the icon for bottles flickered on and off but I have seen the flickering happen with other things that aren't bottles

**Comment by ExterminatorX99** [comment]
Does this error happen on 32bit tModLoader?

**Comment by internetyh** [comment]
Yep, its consistently reproduceable by taking all the bottles out of my storage, and trying to craft a bottle from sand on both 32bit and 64bit
